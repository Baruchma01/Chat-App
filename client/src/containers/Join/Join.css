:root {
    --dark-color-a: #667aff;
    --dark-color-b: #7386ff;
    --light-color: #e6e9ff;
    --success-color: #5cb85c;
    --error-color: #d9534f;
  }


.JoinContainer {
    max-width: 500px;
    margin: 80px auto;
    color: #fff;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
    background: darkcyan;
    border-radius: 20px;
    
}

.Form {
    display: flex;
    flex-direction: column;
    align-items: center;
}

.JoinHeader {
    text-align: center;
    border-top-left-radius: 5px;
    border-top-right-radius: 5px;
    width: 100%;
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
}

.JoinMain {
    padding: 30px 40px;
    width: 100%;
    display: flex;
    width: 100%;
    flex-direction: column;
}

.JoinMain p {
    margin-bottom: 20px;
}

.FormControl{
    margin-bottom: 20px;
    width: 90%;
}

.Label{
    display: block;
    margin-bottom: 5px;
}

.Input {
    font-size: 16px;
    margin: 0;
    padding: 0px;
    height: 39px;
    width: 99%;
}

.RoomSelect {
    font-size: 16px;
    padding: 5px;
    height: 40px;
    width: 100%;
}


.JoinBtn {
    margin-top: 20px;
    cursor: pointer;
    padding: 10px 14px;
    background: var(--light-color);
    color: darkcyan;
    border: 0;
    font-size: 17px;
    border-radius: 7px;
}