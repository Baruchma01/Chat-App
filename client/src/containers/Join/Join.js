import React, { useState, useEffect } from "react";
import './Join.css';
import { useHistory } from "react-router-dom";

const Join = () => {
  const history = new useHistory();
  const [joinFormField, setJoinFormField] = useState({
    username: "",
    room: "JavaScript",
  });

  const changeFormHandler = (e) => {
    setJoinFormField({
      ...joinFormField,
      [e.target.name]: e.target.value,
    });
  };

  const joinChatHandler = (e) => {
    e.preventDefault();
    if (joinFormField.username === "") {
      alert("אנא מלא את שדה השם");
    } else {
      history.push('/chat', {
        user: joinFormField
      });
    }
  };



  return (
    <div className='JoinContainer'>
      <header className='JoinHeader'>
        <h1>
          <i className="fas fa-comments"></i> DeveloperChat
        </h1>
      </header>
      <main className='JoinMain'>
        <form className='Form'>
          <div className='FormControl'>
            <label className='Label'>NickName</label>
            <input
              className='Input'
              type="text"
              name="username"
              id="username"
              placeholder="Enter nickname..."
              required
              value={joinFormField.userName}
              onChange={changeFormHandler}
            />
          </div>
          <div className='FormControl'>
            <label className='Label'>Choose Room</label>
            <select
              className='RoomSelect'
              name="room"
              id="room"
              value={joinFormField.room}
              onChange={changeFormHandler}
            >
              <option value="JavaScript">JavaScript</option>
              <option value="Python">Python</option>
              <option value="PHP">PHP</option>
              <option value="C#">C#</option>
              <option value="Ruby">Ruby</option>
              <option value="Java">Java</option>
            </select>
          </div>
          <button className='JoinBtn' onClick={joinChatHandler}>Join Chat</button>
        </form>
      </main>
    </div>
  );
};

export default Join;
