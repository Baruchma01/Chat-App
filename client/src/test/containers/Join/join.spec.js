import * as React from "react";
import { shallow, mount } from "enzyme";
import Join from "../../../containers/Join/Join";

const mockHistoryPush = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe("MyComponent", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Join />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render Join component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render select,input and button", () => {
    expect(wrapper.find("select")).toHaveLength(1);
    expect(wrapper.find("input")).toHaveLength(1);
    expect(wrapper.find("button")).toHaveLength(1);
  });

  it("should show alert message when username empty", () => {
    window.alert = jest.fn();
    wrapper.find("button").simulate("click");
    expect(window.alert).toBeCalledWith('Name is missed');
  });

  it("should navigate with history push when username are not empty", () => {
    wrapper = mount(<Join />);
    wrapper.find('input.Input').simulate('change', { target: { name: 'username', value: 'Baruch' } } );
    wrapper.find("button").simulate("click");
    expect(mockHistoryPush).toBeCalled();
  })
});
