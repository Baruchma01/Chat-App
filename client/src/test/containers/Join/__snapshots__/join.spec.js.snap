// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render Join component 1`] = `
<div
  className="JoinContainer"
>
  <header
    className="JoinHeader"
  >
    <h1>
      <i
        className="fas fa-comments"
      />
       DeveloperChat
    </h1>
  </header>
  <main
    className="JoinMain"
  >
    <form
      className="Form"
    >
      <div
        className="FormControl"
      >
        <label
          className="Label"
        >
          NickName
        </label>
        <input
          className="Input"
          id="username"
          name="username"
          onChange={[Function]}
          placeholder="Enter nickname..."
          required={true}
          type="text"
          value=""
        />
      </div>
      <div
        className="FormControl"
      >
        <label
          className="Label"
        >
          Choose Room
        </label>
        <select
          className="RoomSelect"
          id="room"
          name="room"
          onChange={[Function]}
          value="JavaScript"
        >
          <option
            value="JavaScript"
          >
            JavaScript
          </option>
          <option
            value="Python"
          >
            Python
          </option>
          <option
            value="PHP"
          >
            PHP
          </option>
          <option
            value="C#"
          >
            C#
          </option>
          <option
            value="Ruby"
          >
            Ruby
          </option>
          <option
            value="Java"
          >
            Java
          </option>
        </select>
      </div>
      <button
        className="JoinBtn"
        onClick={[Function]}
      >
        Join Chat
      </button>
    </form>
  </main>
</div>
`;
